src/test/java/teammates/test/cases/action/StudentProfilePictureDeleteActionTest.java

```diff
@@ -0,0 +1,43 @@
+package teammates.test.cases.action;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.AccountAttributes;
+import teammates.common.util.Const;
+import teammates.ui.controller.RedirectResult;
+import teammates.ui.controller.StudentProfilePictureDeleteAction;
+
+/**
+ * SUT: {@link StudentProfilePictureDeleteAction}.
+ */
+public class StudentProfilePictureDeleteActionTest extends BaseActionTest {
+
+    @Override
+    protected String getActionUri() {
+        return Const.ActionURIs.STUDENT_PROFILE_PICTURE_DELETE;
+    }
+
+    @Override
+    protected StudentProfilePictureDeleteAction getAction(String... params) {
+        return (StudentProfilePictureDeleteAction) gaeSimulation.getActionObject(getActionUri(), params);
+    }
+
+    @Override
+    @Test
+    protected void testExecuteAndPostProcess() throws Exception {
+        AccountAttributes student = typicalBundle.accounts.get("student2InCourse1");
+        String expectedUrl = getPageResultDestination(Const.ActionURIs.STUDENT_PROFILE_PAGE, false, student.googleId);
+        gaeSimulation.loginAsStudent(student.googleId);
+        StudentProfilePictureDeleteAction action = getAction(new String[0]);
+        RedirectResult result = getRedirectResult(action);
+
+        assertEquals(expectedUrl, result.getDestinationWithParams());
+
+    }
+
+    @Override
+    @Test
+    protected void testAccessControl() throws Exception {
+        verifyAnyRegisteredUserCanAccess(new String[0]);
```

Comment:
This does not look safe. So anyone can delete anyone's profile picture?

Reply:
No, only logged in students can delete their own profile picture.
```
    protected void verifyAnyRegisteredUserCanAccess(String[] submissionParams) {
        verifyUnaccessibleWithoutLogin(submissionParams);
        verifyUnaccessibleForUnregisteredUsers(submissionParams);
        verifyAccessibleForStudents(submissionParams);
        verifyAccessibleForAdminToMasqueradeAsStudent(submissionParams);
    }
```
Also we are using this same access control test for `StudentProfilePictureEditActionTest` and `StudentProfilePictureActionTest`.
Why is it unsafe?

Reply:
```java
verifyAccessibleForStudents(submissionParams);
```
Accessible to any student not him/herself? That doesn't sound right; you don't want other people to delete your profile picture, right?

Reply:
But when a student logs in, how will he/her get access to other student's profile picture?
Doesn't this test mean that this action can only be used by registered students?

Reply:
i had a look at `StudentFeedbackSubmissionEditSaveActionTest` to understand how we are testing Access Control of an individual.
```
    protected void testAccessControl() throws Exception {
        FeedbackResponseAttributes fr = typicalBundle.feedbackResponses.get("response1ForQ1S1C1");

        String[] submissionParams = new String[] {
                Const.ParamsNames.FEEDBACK_SESSION_NAME, fr.feedbackSessionName,
                Const.ParamsNames.COURSE_ID, fr.courseId,
                Const.ParamsNames.FEEDBACK_QUESTION_ID + "-1", fr.feedbackQuestionId,
                Const.ParamsNames.FEEDBACK_RESPONSE_RECIPIENT + "-1-0", fr.recipient,
                Const.ParamsNames.FEEDBACK_QUESTION_TYPE + "-1", fr.feedbackQuestionType.toString(),
                Const.ParamsNames.FEEDBACK_RESPONSE_TEXT + "-1-0", fr.getResponseDetails().getAnswerString()
        };

        verifyOnlyStudentsOfTheSameCourseCanAccess(submissionParams);
        testGracePeriodAccessControlForStudents();
    }
```
Now ` verifyOnlyStudentsOfTheSameCourseCanAccess(submissionParams);` doesn't mean that any student of same course can edit fs of some other student?

Reply:
@wkurniawan07 Can you please point out what should be used instead of `verifyAccessibleForStudents(submissionParams)` ? I am not able to figure it out. Thanks.